### DISCUSSION TITLE
Which architecture is better for training ?

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#1757] [General Q&A] [answered]
Hi,

I am willing to build a double gpu arch to train a VITS  model.

Would it be better to pair **identical** gpus (eg 2 x gtx 1660 super with 6GiB) or it does not matter if they are not identical, the quickest gpus would also be the quickest to finish training (eg pairing rtx 3070 with rtx 3060)?

Any hint greatly appreciated

**Comment by p0p4k** [comment]
Try doing a mini training (few epochs) in both setups and find out yourself :)

**Reply by Ca-ressemble-a-du-fake** [reply]
Correct "probieren geht uber studieren" (Testing is above studying) ! Though I wanted to know if there were some theory that I could verify on the field.

**Comment by Ca-ressemble-a-du-fake** [comment, marked as answer]
While still researching on that topic (because I am willing to avoid unnecessary  swapping of (expensive) GPUs from one place to another one) I found this answer from https://www.reddit.com/r/buildapc/comments/rx8e11/does_anybody_run_a_dual_geforce_rtx_3060/

Which states _All DL frameworks have multi gpu support, so you can deal [w]ith it on code level. _

So this would mean that pairing _identical_ GPUs is not required for training since it is handled by the DL framework. 

And the DL framework in use here is PyTorch which documentation (https://pytorch.org/tutorials/beginner/former_torchies/parallelism_tutorial.html) does not mention the need to use _identical_ GPUs. 

Consequently in my case it seems that pairing the most powerful GPUs will give the best results even though they are different.

**Reply by p0p4k** [reply]
Good find. Thanks for sharing!

**Reply by Ca-ressemble-a-du-fake** [reply]
I could test one setup today (gtx 1660 super paired with rtx 3070 on the same motherboard without nvlink). And a new issue occured. When running in single gpu mode with gtx 1660 the training runs. But when trying to run in single mode with rtx, I get CUDA error because rtx needs at least "sm_80".

So it looks like those GPUs aren't from the same gpu architecture (AMPERE, TURING,...) and so they don't have the same "sm" version.

I tried to swap GPUs on the motherboard so that rtx 3070 is first but it did not change.

Consequently, so far I wasn't able to test multi gpus gtx-rtx. I will try to install new drivers via the GUI now that the first gpu is rtx and see if "sm" version increases.
